docs/development.rst

```diff
@@ -148,43 +148,65 @@ Workflow
+    proposed addition should be added to the codebase, proceed!
+
 1. Make your own `fork <https://help.github.com/articles/fork-a-repo/>`_
    of the ``tskit`` repository on GitHub, and
    `clone <https://help.github.com/articles/cloning-a-repository/>`_
    a local copy as detailed in :ref:`sec_development_getting_started_environment`.
+
 2. Make sure that your local repository has been configured with an
    `upstream remote <https://help.github
    .com/articles/configuring-a-remote-for-a-fork/>`_::
 
    $ git remote add upstream git@github.com:tskit-dev/tskit.git
 
+
 3. Create a "topic branch" to work on. One reliable way to do it
    is to follow this recipe::
 
     $ git fetch upstream
     $ git checkout upstream/main
     $ git checkout -b topic_branch_name
 
-4. As you work on your topic branch you can add commits to it. Once you're
+
+4. Write your code following the outline in :ref:`sec_development_best_practices`.
+   As you work on your topic branch you can add commits to it. Once you're
    ready to share this, you can then open a `pull request
-   <https://help.github.com/articles/about-pull-requests/>`__. Your PR will
-   be reviewed by some of the maintainers, who may ask you to make changes.
-   If you'd like to open the PR but feel that the code isn't ready
-   for review yet, please use the "Draft" option on GitHub.
-5. If your topic branch has been around for a long time and has gotten
-   out of date with the main repository, we might ask you to
-   `rebase <https://help.github.com/articles/about-git-rebase/>`_
-   or to "squash" your commits into one.
+   <https://help.github.com/articles/about-pull-requests/>`__. This can be done at any time
+   if you would like feedback on incomplete code, the drop-down button to create a PR
+   has a "draft PR" option for this.
+
+5. As you code it is best to `rebase <https://stdpopsim.readthedocs.io/en/
+   latest/development.html#rebasing>`_ your work onto the ``main`` branch periodically
+   (e.g. once a week) to keep up with changes.
+
+6. Once you're done coding add content to the tutorial and other documentation pages if
+   appropriate.
 
-Please follow `this guide
-<https://stdpopsim.readthedocs.io/en/stable/development.html#rebasing>`_
-for step-by-step instructions on rebasing and squashing commits.
+7. Update the change logs, taking care to document any breaking changes separately.
+
+8. Push your changes to your topic branch and either open the PR or, if you
+   opened a draft PR above change it to a non-draft PR by clicking "Ready to
+   Review".
+
+9. The tskit community will review the code, asking you to make changes where appropriate.
+   This can often take a couple of rounds of review.
+
+10. Once the review process is complete squash the commits to the minimal set of changes -
+    usually one or two. Please follow `this guide <https://stdpopsim.readthedocs.io/en/stable/development.html#rebasing>`_ for step-by-step
```

Comment:
"usually one or two commits"

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/915/commits/68c03c53e0f671a3728db678a16908d02bfedea0